### PR TITLE
[tests-only] Adjust groupTest interactions for oCIS server

### PR DESCRIPTION
**Pull request by saw-jan**
part of https://github.com/owncloud/owncloud-sdk/issues/644 and https://github.com/owncloud/owncloud-sdk/issues/747

adjust interactions for oCIS:
- groupTest.js

### CODE REVIEW

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/groupTest.js [hunk 1]
```diff
@@ -138,28 +141,41 @@ describe('Main: Currently testing group management,', function () {
...
   })
 
   it('checking method : deleteGroup with a non-existent group', async function () {
-    await deleteGroupInteraction(provider, config.nonExistentGroup)
+    await deleteGroupInteraction(
+      provider,
+      config.nonExistentGroup,
+      meta => ocsMeta(
+        meta,
+        MatchersV3.regex('(failure|error)', 'failure'),
```
Comment (individual-it) [inline tests/groupTest.js:174]: is this a difference in ocis vs. oc10? is there an issue open for that?

File: tests/groupTest.js [hunk 2]
```diff
@@ -32,19 +32,22 @@ describe('Main: Currently testing group management,', function () {
         headers: xmlResponseHeaders,
         body: new XmlBuilder('1.0', '', 'ocs').build(ocs => {
           ocs.appendElement('meta', '', (meta) => {
-            return ocsMeta(meta, 'ok', '100')
+            return ocsMeta(meta, 'ok', '100', MatchersV3.regex('(OK)?', ''))
```
Comment (individual-it) [inline tests/groupTest.js:35]: does the client need this response, if not we should ignore it altogether and never send it back

File: tests/groupTest.js [hunk 3]
```diff
@@ -88,7 +89,6 @@ describe('Main: Currently testing group management,', function () {
       await oc.login()
       return oc.groups.getGroups().then(data => {
         expect(typeof (data)).toBe('object')
-        expect(data.indexOf('admin')).toBeGreaterThan(-1)
```
Comment (individual-it) [inline tests/groupTest.js:91]: It does not feel good for me to remove this check
maybe we can add the "admin" group in ocis

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/groupTest.js (hunk 2 above)
Comment (saw-jan) [reply tests/groupTest.js:35]: It seems that client needs `meta.message` too. If we are to append `message` element only if needed like this
```js
if (Message !== null) {
    meta.appendElement('message', '', Message)
}
```
then the consumer tests will fail with following error
```js
TypeError: Cannot convert undefined or null to object
        at Function.keys (<anonymous>)

      392 |       ret = meta.message
      393 | 
    > 394 |       if (Object.keys(meta.message).length === 0) {
          |                  ^
      395 |         // no error message returned, return the whole message
      396 |         ret = json
      397 |       }
```
At least empty `message` element needs to be appended

**Review by individual-it — APPROVED** [review]

**Pull request merged**